vnext/Shared/FeatureGate.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <string>
+
+namespace Microsoft::React {
+
+void SetFeatureGate(std::string &&name) noexcept;
```

Comment:
>void [](start = 0, length = 4)

Please add detailed comments on how these apis should be used. Eg: single threaded requirement, called during initialization, etc